src/drawtypes/label.cpp

```diff
@@ -73,6 +82,28 @@ namespace drawtypes {
     else
       text = conf.get<string>(section, name, def);
 
```

Comment:
Noticed some errors in the token parsing block. Here's a proposed patch:

- Skip all non-token values
- Support `%token:<min>%`
- Catch exception thrown on value conversion failure

~~~ cpp
size_t start, end, pos;
string line{text};

while ((start = line.find('%')) != string::npos && (end = line.find('%', start + 1)) != string::npos) {
  auto token = line.substr(start, end);

  line.erase(0, end);

  // ignore false positives (lemonbar-style declarations)
  if (token[1] == '{')
    continue;

  bound.emplace_back(bounds{0, 0});

  // find min delimiter
  if ((pos = token.find(':')) == string::npos)
    continue;

  // strip min/max specifiers from the label string token
  text = string_util::replace(text, token, token.substr(0, pos));

  try {
    bound.back().min = std::stoul(&token[pos + 1], nullptr, 10);
  } catch (const std::invalid_argument& err) {
    continue;
  }

  // find max delimiter
  if ((pos = token.find(':', pos + 1)) == string::npos)
    continue;

  try {
    bound.back().max = std::stoul(&token[pos + 1], nullptr, 10);
  } catch (const std::invalid_argument& err) {
    continue;
  }

  // ignore max lengths less than min
  if (bound.back().max < bound.back().min)
    bound.back().max = 0;
}
~~~